src/main/protobuf/bazel_output_service.proto

```diff
@@ -0,0 +1,324 @@
+  // Content Addressable Storage. Contents of these files can be lazily staged.
+  rpc StageArtifacts(StageArtifactsRequest) returns (StageArtifactsResponse);
+
+  // Notifies server that a set of files will no longer be modified by Bazel or
+  // local build actions. Server should unset "dirty" bit for these files if it
+  // tracks file modifications under output path. If these paths are modified
+  // via FUSE operations later, they should be reported back to Bazel in the
+  // next InitialOutputPathContents.modified_paths for the same workspace.
+  rpc FinalizeArtifacts(FinalizeArtifactsRequest) returns (FinalizeArtifactsResponse);
+
+  // Signal that a build has been completed.
+  rpc FinalizeBuild(FinalizeBuildRequest) returns (FinalizeBuildResponse);
+
+  // Obtain the status of one or more files, directories or symbolic
+  // links that are stored in the output path.
+  rpc BatchStat(BatchStatRequest) returns (BatchStatResponse);
+}
+
+message CleanRequest {
+  // The workspace identifier that was provided to
+  // StartBuildRequest.output_base_id whose data needs to be removed.
+  string output_base_id = 1;
+}
+
+message CleanResponse {
+  // Intentionally empty for future extensibility.
+}
+
+message StartBuildRequest {
+  // The version of the protocol Bazel currently uses. The service MUST return
+  // an error if it doesn't recognize the version.
+  //
+  // A future Bazel version may introduce incompatible changes and increment
+  // this version number. The incompatible change will be first made in the
+  // development tree for the next major Bazel release, and the new version thus
+  // introduced should be considered unstable until that major Bazel release
+  // occurs. At that time, the new version becomes stable, and the old one is
+  // immediately retired.
+  //
+  // In particular, version 1 must not be considered stable until Bazel 8.0.0 is
+  // released.
+  //
+  // Current version: 1 (experimental).
+  int32 version = 1;
+
+  // A client-chosen value for the output service to uniquely identify the
+  // workspace the build is being started. This value must be set to ensure that
+  // the remote output service is capable of managing builds for distinct
+  // workspaces concurrently.
+  //
+  // Bazel sets this value to the MD5 sum of the absolute path of the output
+  // base.
+  //
+  // Starting a build finalizes any previous build with the same output_base_id
+  // that has not been finalized yet as if a FinalizeBuildRequest had been sent
+  // with build_successful set to false.
+  string output_base_id = 2;
+
+  // A client-chosen value that uniquely identifies this build. This value must
+  // be provided to most other methods to ensure that operations are targeted
+  // against the right output path.
+  //
+  // Bazel sets this value to --invocation_id.
+  string build_id = 3;
+
+  // Additional arguments to pass depending on how Bazel communicate with the
+  // Content Addressable Storage.
```

Comment:
Suggest adding "In case of a REv2 CAS, the type is ...", mirroring the `digest` field.

Reply:
Done.